### PR TITLE
Feat/implement progress screen UI

### PR DESCRIPTION
**Pull request by Alex720p**
Screenshot 2024-11-08 161213

- **`Progression.kt`**
  - Implemented Progress UI screen following the figma design (except for the metric cards that I centered, as for smaller screen the rows collapse and center themself, this keeps the design responsive). All the values are placeholders, will connect the viewmodel in later pr.

- **`NavigationActions.kt:`**
  -  Changed `TopLevelDestinations` class to allow for ressources images to be used aswell.
  - Added the Progression screen, route and added them to the global lists.
  - Setup the Progression `ScreenParams` params and add it to the global list.

- **`BottomNavigationMenu.kt:`**
  -  Changed `BottomNavigationMenu` composable to allow for ressources images to be used as icons from `TopLevelDestination` class.
  - Resized and changed the color to make it match the figma color palette.

- **`TopAppBar.kt:`**
  - Adapted the color to match the figma choices.

- **`Color.kt:`**
  - Added a `ColorPalette` object that references all the colors from the figma.

### CODE REVIEW

File: app/src/main/java/com/android/streetworkapp/ui/progress/Progression.kt [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.android.sample.R
+import com.android.streetworkapp.ui.navigation.NavigationActions
+import com.android.streetworkapp.ui.theme.ColorPalette
+
+// TODO: we need to define a global color palette somewhere
+val PROGRESSION_COLOR_BLUE = Color(0xFF007BFF)
+val PROGRESSION_COLOR_GRAY = Color(0xFFDDDDDD)
```
Comment (tercierp) [inline app/src/main/java/com/android/streetworkapp/ui/progress/Progression.kt:50]: Question: Did you forget to delete the todo or do you still need to change something?

File: app/src/main/java/com/android/streetworkapp/ui/progress/Progression.kt [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+
+        Spacer(modifier = Modifier.width(16.dp))
+
+        Column(modifier = Modifier.weight(1f)) {
+          // Achievement Title
+          Text(
+              text = achievement.title,
+              fontSize = 16.sp,
+              fontWeight = FontWeight.Bold,
+              color = ColorPalette.PRIMARY_TEXT_COLOR)
```
Comment (tercierp) [inline app/src/main/java/com/android/streetworkapp/ui/progress/Progression.kt:265]: Suggestion: add `maxLines = 1`  and `overflow = TextOverflow.Ellipsis` to counter any overflow

File: app/src/main/java/com/android/streetworkapp/ui/progress/Progression.kt [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+
+          // Tags Row
+          FlowRow(
```
Comment (tercierp) [inline app/src/main/java/com/android/streetworkapp/ui/progress/Progression.kt:268]: Suggestion: I think that `LazyRow` is more performant in this context

File: app/src/main/java/com/android/streetworkapp/ui/progress/Progression.kt [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+              horizontalArrangement = Arrangement.spacedBy(4.dp),
+              verticalArrangement = Arrangement.spacedBy(5.dp),
+              modifier = Modifier.padding(top = 4.dp)) {
+                achievement.tags.forEach { tag ->
+                  Text(
+                      text = tag,
+                      fontSize = 12.sp,
+                      color = ColorPalette.SECONDARY_TEXT_COLOR,
+                      modifier =
+                          Modifier.background(Color(0xFFE0E0E0), shape = RoundedCornerShape(4.dp))
```
Comment (tercierp) [inline app/src/main/java/com/android/streetworkapp/ui/progress/Progression.kt:278]: Improvement: use a color in the palette `Color.kt` or define for a good reason a new color in the palette

File: app/src/main/java/com/android/streetworkapp/ui/progress/Progression.kt [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+    append("780")
+
+    withStyle(style = SpanStyle(fontWeight = FontWeight.SemiBold)) { append("/1000") }
+  }
+
+  Box(modifier = Modifier.padding(paddingValues).testTag("progressionScreen")) {
+    LazyColumn(
+        modifier =
+            Modifier.fillMaxSize()
+                .padding(PaddingValues(0.dp, progressBarSize * 0.15f, 0.dp, 0.dp)),
```
Comment (tercierp) [inline app/src/main/java/com/android/streetworkapp/ui/progress/Progression.kt:124]: Suggestion: this is not a huge performance win but every recompositions this value is recomputed
You could mitigate this problem like this:
```kotlin
val columnPadding = PaddingValues(0.dp, Dimens.ProgressBarSize * 0.15f, 0.dp, 0.dp)
LazyColumn(
    modifier =
        Modifier.fillMaxSize().padding(columnPadding),
)
```

**Review by tercierp — CHANGES_REQUESTED** [review]
The result is very faithful to the Figma, well done!
There are a few suggestions and improvements I've made, I'll let you answer and/or fix them.

File: app/src/main/java/com/android/streetworkapp/ui/progress/Progression.kt (hunk 1 above)
Comment (Alex720p) [reply app/src/main/java/com/android/streetworkapp/ui/progress/Progression.kt:50]: yep forgot it, removed it

File: app/src/main/java/com/android/streetworkapp/ui/progress/Progression.kt (hunk 2 above)
Comment (Alex720p) [reply app/src/main/java/com/android/streetworkapp/ui/progress/Progression.kt:265]: Screenshot 2024-11-09 082523


personally not a fan of having the end of text being hidden. I think keeping it as is is cleaner.

File: app/src/main/java/com/android/streetworkapp/ui/progress/Progression.kt (hunk 3 above)
Comment (Alex720p) [reply app/src/main/java/com/android/streetworkapp/ui/progress/Progression.kt:268]: Yes but this wouldn't allow for the tags to be on multiple lines. Keeping the FlowRow will keep the design more responsive.

File: app/src/main/java/com/android/streetworkapp/ui/progress/Progression.kt (hunk 4 above)
Comment (Alex720p) [reply app/src/main/java/com/android/streetworkapp/ui/progress/Progression.kt]: Screenshot 2024-11-09 083331


changed it to `ColorPalette.BORDER_COLOR`

File: app/src/main/java/com/android/streetworkapp/ui/progress/Progression.kt (hunk 5 above)
Comment (Alex720p) [reply app/src/main/java/com/android/streetworkapp/ui/progress/Progression.kt]: done

File: app/src/main/java/com/android/streetworkapp/ui/progress/Progression.kt (hunk 2 above)
Comment (tercierp) [reply app/src/main/java/com/android/streetworkapp/ui/progress/Progression.kt:268]: Yep does make sense resolved

File: app/src/main/java/com/android/streetworkapp/ui/progress/Progression.kt (hunk 3 above)
Comment (tercierp) [reply app/src/main/java/com/android/streetworkapp/ui/progress/Progression.kt:271]: Given your argument conversation resolved

**Review by tercierp — APPROVED** [review]
Thank you for your changes and your explanation

**Pull request merged**